revocation/ocsp/ocsp.go

```diff
@@ -78,16 +71,14 @@ func CheckStatus(opts Options) ([]*result.CertRevocationResult, error) {
 	// Thus, it is better to pass nil here than fail for a cert's NotBefore
 	// being after zero time
 	switch opts.CertChainPurpose {
-	case PurposeCodeSigning:
-		if err := coreX509.ValidateCodeSigningCertChain(opts.CertChain, nil); err != nil {
-			return nil, result.InvalidChainError{Err: err}
-		}
-	case PurposeTimestamping:
+	case x509.ExtKeyUsageTimeStamping:
 		if err := coreX509.ValidateTimestampingCertChain(opts.CertChain); err != nil {
 			return nil, result.InvalidChainError{Err: err}
 		}
 	default:
-		return nil, result.InvalidChainError{Err: fmt.Errorf("unknown certificate chain purpose %v", opts.CertChainPurpose)}
+		if err := coreX509.ValidateCodeSigningCertChain(opts.CertChain, nil); err != nil {
+			return nil, result.InvalidChainError{Err: err}
+		}
```

Comment:
Setting it to default is dangerous.

Can you modify code and doc saying that setting to x509.ExtKeyUsageAny (which is `0`) implies `CodeSigning`?

Reply:
Updated.